### PR TITLE
Refactor the definition of monads

### PR DESCRIPTION
**Pull request by EgbertRijke**
This PR breaks up the definition of a monad into smaller parts.

### CODE REVIEW

**Comment by EgbertRijke** [comment]
I have added everything I wanted to add in this PR, so it is ready for review.

File: src/category-theory/monads-on-categories.lagda.md [hunk 1]
```diff
@@ -8,23 +8,239 @@ module category-theory.monads-on-categories where
...
 open import foundation.universe-levels
+
+open import foundation-core.cartesian-product-types
 ```
 
 </details>
 
+## Idea
+
+A monad on a category `C` consists of an
```
Comment (fredrik-bakke) [inline src/category-theory/monads-on-categories.lagda.md:27]:
```suggestion
A {{#concept "monad" Disambiguation="on a category" Agda=monad-Category}} on a [category](category-theory.categories.md) `C` consists of an
```

File: src/category-theory/pointed-endofunctors-categories.lagda.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+open import foundation.dependent-pair-types
+open import foundation.identity-types
+open import foundation.universe-levels
+```
+
+</details>
+
+## Idea
+
+An [endofunctor](category-theory.functors-categories.md) `F : C → C` on a
```
Comment (fredrik-bakke) [inline src/category-theory/pointed-endofunctors-categories.lagda.md:24]:
```suggestion
An endo[functor](category-theory.functors-categories.md) `F : C → C` on a
```

File: src/category-theory/pointed-endofunctors-precategories.lagda.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+open import foundation.dependent-pair-types
+open import foundation.identity-types
+open import foundation.universe-levels
+```
+
+</details>
+
+## Idea
+
+An [endofunctor](category-theory.functors-precategories.md) `F : C → C` on a
```
Comment (fredrik-bakke) [inline src/category-theory/pointed-endofunctors-precategories.lagda.md:23]:
```suggestion
An endo[functor](category-theory.functors-precategories.md) `F : C → C` on a
```

File: src/category-theory/pointed-endofunctors-precategories.lagda.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+       η X
+    X -----> F X
+    |         |
+  f |         | F f
+    ∨         ∨
+    Y -----> F Y
+       η Y
+```
+
+commutes.
```
Comment (fredrik-bakke) [inline src/category-theory/pointed-endofunctors-precategories.lagda.md:46]:
```suggestion
[commutes](category-theory.commuting-squares-of-morphisms-in-precategories.md)
```

**Review by fredrik-bakke — APPROVED** [review]
Looks good to me. Perhaps @GregorPercic wants to have a look at this as well?

**Comment by GregorPercic** [comment]
Looks excellent to me. :)

File: src/category-theory/pointed-endofunctors-categories.lagda.md (hunk 2 above)
Comment (fredrik-bakke) [reply src/category-theory/pointed-endofunctors-categories.lagda.md:24]: Oh, you did so elsewhere in the PR

File: src/category-theory/monads-on-categories.lagda.md [hunk 5]
```diff
@@ -8,23 +8,239 @@ module category-theory.monads-on-categories where
...
+endo[functor](category-theory.functors-categories.md) `T : C → C` together with
```
Comment (fredrik-bakke) [inline src/category-theory/monads-on-categories.lagda.md:28]:
```suggestion
[endofunctor](category-theory.functors-categories.md) `T : C → C` together with
```

File: src/category-theory/pointed-endofunctors-categories.lagda.md (hunk 2 above)
Comment (EgbertRijke) [reply src/category-theory/pointed-endofunctors-categories.lagda.md:24]: Ah, I must have been sloppy. Thanks for pointing that out.

**Pull request merged**